scripts/zones/Balgas_Dais/mobs/Wyrm.lua

```diff
@@ -4,5 +4,67 @@
 -- KSNM: Early Bird Catches the Wyrm
 -----------------------------------
 
+function onMobInitialize(mob)
+    mob:setMobMod(tpz.mobMod.DRAW_IN, 1) -- has a bug during flight, like Tiamat
+end
+
+function onMobSpawn(mob)
+    mob:setTP(3000) -- opens fight with a skill
+end
+
+function onMobFight(mob, target)
+    local isBusy = false
+    local action = mob:getCurrentAction()
+    local hasLanded = mob:getLocalVar("hasLanded")
+
+    if action == tpz.act.MOBABILITY_START or action == tpz.act.MOBABILITY_USING or action == tpz.act.MOBABILITY_FINISH then
+        isBusy = true -- when the Wyrm is in any stage of using a mobskill
+    end
```

Comment:
From latest testing I find that checking only for `actionQueueEmpty()` does not prevent overflow. The animation for `AnimationSub(0)` will still overwrite `Touchdown`. This function is called really fast. I propose a mixed use, which from my testing seems to work as expected (Touchdown finishes and then stays on ground). This way I can still remove all the later checks.

```LUA
    -- Testing for actionQueueEmpty() does not prevent overflow
    if action == tpz.act.MOBABILITY_START or action == tpz.act.MOBABILITY_USING or action == tpz.act.MOBABILITY_FINISH then
        isBusy = true -- when the Wyrm is in any stage of using a mobskill
    end
    -- Do nothing if currently doing something else already
    if isBusy then
        return
    end
```

Reply:
I had similar probs with other mobs I worked on that changed animation states. What Kjlotus told me was the action queue is basically _never_ empty. I suggest turning it into a local function instead of defined inside onMobFight, and then you can just use it as the condition before other logic executes instead of returning out. `if notBusy() then`

Reply:
I did as you suggested and outsourced this into its own local function for re-use in later conditions. Works well.